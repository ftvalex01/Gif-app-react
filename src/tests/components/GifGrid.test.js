import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en GifGrid', () => {
    
    const category = 'One Punch';
    

    test('Debe mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()

    });

    test('debe de mostrar items cuando carga las imagenes con useFetch', () => {
        
       const gifs =[{
           id:'ABC',
           url:'https://localhost/cualquier/cosa.jpg',
           title:'Cualquier cosa'
       },
       {
        id:'11ABC',
        url:'https://localhost/cualquier/cosa.jpg',
        title:'Cualquier cosa'
        }];
       useFetchGifs.mockReturnValue({
        data:gifs,
        loading:false
    });

       const wrapper = shallow(<GifGrid category={category}/>);
       
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('p').exists()).toBe(false);
       expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    });

});