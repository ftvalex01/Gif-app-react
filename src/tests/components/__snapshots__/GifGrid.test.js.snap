// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid debe de mostrar items cuando carga las imagenes con useFetch 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="11ABC"
      key="11ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
